Userland/Libraries/LibWeb/HTML/WorkerLocation.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2022, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/String.h>
+#include <LibWeb/HTML/WorkerGlobalScope.h>
+#include <LibWeb/HTML/WorkerLocation.h>
+
+namespace Web::HTML {
+
+// https://html.spec.whatwg.org/multipage/workers.html#worker-locations
```

Comment:
Can we add a link to each individual function instead?

```
https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-href
https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-origin
https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-protocol
https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-host
https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-hostname
https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-port
https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-pathname
https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-search
https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-hash
```

Reply:
(got these by clicking on each name in bold font)

Reply:
Yep added throughout. maybe a bit overboard but I'm sure i'll get more comments if that's the case.

Reply:
Not overboard at all. If it has comments with steps from the spec, it needs a link :)